### DISCUSSION TITLE
[Issue]:Scoring DNA damage on cell nucleus

### DISCUSSION THREAD

**Discussion by LucasHenriqueDG** [topas-nbio/TOPAS-nBio#87] [Simulation-question]
### What was the issue?

I'm trying to run some simulations, but I've faced some issues related to the DNADamageNucleusStepByStep.
I've successfully made some simulations using the TsNucleus, however, when i try to score the DNA damage of a Nucleus inside a Cell, it doesn't work, is that possible or not?

My Code (I'm adapting the examples):
`# -- Geometry
# World
d:Ge/World/HLX                          = 15.0 um
d:Ge/World/HLY                          = 15.0 um
d:Ge/World/HLZ                          = 15.0 um
b:Ge/World/Invisible                    = "true"
s:Ge/World/Material                     = "G4_WATER"

s:Ge/MyCell/Type="TsSphericalCell"
s:Ge/MyCell/Material="G4_WATER"
s:Ge/MyCell/Parent="World"
d:Ge/MyCell/CellRadius=10 um
s:Ge/MyCell/Color="white"

s:Ge/MyCell/Nucleus/Type                   = "TsNucleus"
s:Ge/MyCell/Nucleus/Material                   = "G4_WATER"
s:Ge/MyCell/Nucleus/Color                      = "RED"
s:Ge/MyCell/Nucleus/DNAModel					= "Sphere"
d:Ge/MyCell/Nucleus/NucleusRadius              = 4.65 um
i:Ge/MyCell/Nucleus/HilbertCurveLayer          = 4
i:Ge/MyCell/Nucleus/HilbertCurve3DRepeat       = 30
s:Ge/MyCell/Nucleus/HilbertCurveFileName       = "supportFiles/HilbertPoints_iteration1.dat"
s:Ge/MyCell/Nucleus/Voxel/Material             = "G4_WATER"
b:Ge/MyCell/Nucleus/ShowDNAVoxels              = "false"
s:Ge/MyCell/Nucleus/ChromatinFiber/Material    = "G4_WATER"
b:Ge/MyCell/Nucleus/ShowChromatinCylinders     = "false"
s:Ge/MyCell/Nucleus/Histone/Material           = "G4_HistoneMaterial"
b:Ge/MyCell/Nucleus/OnlyBuildOneHistone        = "false"
b:Ge/MyCell/Nucleus/HistoneAsScavenger         = "true"
b:Ge/MyCell/Nucleus/AddBases                   = "true"
s:Ge/MyCell/Nucleus/Base1/Material             = "G4_BaseMaterial"
s:Ge/MyCell/Nucleus/Base2/Material             = "G4_BaseMaterial"
b:Ge/MyCell/Nucleus/AddBackbones               = "true"
s:Ge/MyCell/Nucleus/Backbone1/Material         = "G4_BackboneMaterial"
s:Ge/MyCell/Nucleus/Backbone2/Material         = "G4_BackboneMaterial"
b:Ge/MyCell/Nucleus/AddHydrationShell          = "true"
s:Ge/MyCell/Nucleus/HydrationShell1/Material   = "G4_WATER_MODIFIED"
s:Ge/MyCell/Nucleus/HydrationShell2/Material   = "G4_WATER_MODIFIED"

# -- Materials
s:Ma/G4_HistoneMaterial/CloneFromMaterial   = "G4_WATER"
d:Ma/G4_HistoneMaterial/CloneWithDensity    = 1.407 g/cm3
s:Ma/G4_BaseMaterial/CloneFromMaterial      = "G4_WATER"
d:Ma/G4_BaseMaterial/CloneWithDensity       = 1.0 g/cm3
s:Ma/G4_BackboneMaterial/CloneFromMaterial  = "G4_WATER"
d:Ma/G4_BackboneMaterial/CloneWithDensity   = 1.0 g/cm3
s:Ma/G4_WATER_MODIFIED/CloneFromMaterial    = "G4_WATER"
d:Ma/G4_WATER_MODIFIED/CloneWithDensity     = 1.0 g/cm3

# -- Physics and Chemistry
sv:Ph/Default/Modules                      		= 1 "g4em-dna"
#sv:Ph/Default/Modules								= 2 "g4em-dna" "TsEmDNAChemistry" 	# OR	 physics and chemistry
#includeFile											= supportFiles/TOPASDefaultReactions.txt
#s:Ch/ChemistryName									= "TOPASChemistry"
#b:Ch/TOPASChemistry/ChemicalStageTransportActive    = "True"
i:Ph/Verbosity                          			= 0

# -- Source and runs
s:So/ParticleGun/Type                               = "Beam"
s:So/ParticleGun/Component                          = "BeamPosition"
d:Ge/BeamPosition/TransX                            = 0.05 um
d:Ge/BeamPosition/TransY                            = 0.05 um
d:Ge/BeamPosition/TransZ                            = 4.65 um
s:So/ParticleGun/BeamParticle                       = "alpha"
s:So/ParticleGun/BeamPositionDistribution           = "none"					# EITHER single track position
#s:So/ParticleGun/BeamPositionDistribution			= "Flat"					# OR 	 disk source
#s:So/ParticleGun/BeamPositionCutoffShape			= "Ellipse"
#d:So/ParticleGun/BeamPositionCutoffX 				= Ge/MyCell/Nucleus/NucleusRadius um
#d:So/ParticleGun/BeamPositionCutoffY 				= Ge/MyCell/Nucleus/NucleusRadius um
s:So/ParticleGun/BeamAngularDistribution            = "none"
d:So/ParticleGun/BeamEnergy                         = 5.0 MeV
i:So/ParticleGun/NumberOfHistoriesInRun             = 10
b:Ge/MyCell/Nucleus/RotateNucleusForEachRun				= "false"
# Producing separate runs instead of separate histories if rotation enabled
#d:Tf/TimelineStart 			 					= 0. s
#d:Tf/TimelineEnd   			 					= 10. s
#i:Tf/NumberOfSequentialTimes 						= 5 # Number of runs

# -- Scorer --
includeFile									= DNAScorerStepByStep.txt
s:Sc/DNADamageScorer/Quantity				= "DNADamageNucleusStepByStep"
s:Sc/DNADamageScorer/Component 				= "MyCell/Nucleus"
i:Sc/DNADamageScorer/NumberOfHistoriesInRun = So/ParticleGun/NumberOfHistoriesInRun
d:Sc/DNADamageScorer/ScoringRadius			= Ge/MyCell/Nucleus/NucleusRadius um
i:Sc/DNADamageScorer/BasePairPositionAtGeometricHierarchy = 0
sv:Sc/DNADamageScorer/Strand1MaterialNames  = 4 "G4_WATER_MODIFIED" "G4_BackboneMaterial" "G4_BaseMaterial" "G4_HistoneMaterial"
sv:Sc/DNADamageScorer/Strand2MaterialNames  = 3 "G4_WATER_MODIFIED" "G4_BackboneMaterial" "G4_BaseMaterial"
s:Sc/DNADamageScorer/PrimaryParticle		= So/ParticleGun/BeamParticle
d:Sc/DNADamageScorer/MeanEnergy				= So/ParticleGun/BeamEnergy MeV
#d:Sc/DNADamageScorer/StopTrackingAtDose		= 0.5 gray
d:Sc/DNADamageScorer/ImageXMinPosition		= 0 um - Ge/MyCell/Nucleus/NucleusRadius
d:Sc/DNADamageScorer/ImageXMaxPosition		= Ge/MyCell/Nucleus/NucleusRadius um
d:Sc/DNADamageScorer/ImageYMinPosition		= 0 um - Ge/MyCell/Nucleus/NucleusRadius
d:Sc/DNADamageScorer/ImageYMaxPosition		= Ge/MyCell/Nucleus/NucleusRadius um
d:Sc/DNADamageScorer/ImageZMinPosition		= 0 um - Ge/MyCell/Nucleus/NucleusRadius
d:Sc/DNADamageScorer/ImageZMaxPosition		= Ge/MyCell/Nucleus/NucleusRadius um
s:Sc/DNADamageScorer/GeometryInfo = "supportFiles/Geoinfo.txt"
s:Sc/DNADamageScorer/CopyNoTable = "supportFiles/CopyNoTable.txt"
s:Sc/DNADamageScorer/FileVoxelCopyNumbersInsideNucleus = "supportFiles/CopyNoTable.txt"
s:Sc/DNADamageScorer/MapChromosomeVoxelsFile = "supportFiles/signedCHVoxel.txt"
s:Sc/DNADamageScorer/signedCHVoxel = "supportFiles/signedCHVoxel.txt"
Ts/ShowHistoryCountAtInterval                       = 1

#Score dose to cell
s:Sc/CellScorer/Quantity = "DoseToMedium"
s:Sc/CellScorer/Component = "MyCell"
b:Sc/CellScorer/OutputToConsole = "TRUE"
b:Sc/CellScorer/PropagateToChildren = "true"
s:Sc/CellScorer/IfOutputFileAlreadyExists = "overwrite"

#Score in the Cell components
s:Sc/OrganelleScorer/Quantity = "TsNtupleForCell"
s:Sc/OrganelleScorer/Component = "MyCell"
s:Sc/OrganelleScorer/IfOutputFileAlreadyExists = "Overwrite"  
s:Sc/OrganelleScorer/OutputType = "ASCII" 
s:Sc/OrganelleScorer/OutputFile = "EdepToOrganelles"
b:Sc/OrganelleScorer/PropagateToChildren = "true"

# -- Visualization

#s:Gr/ViewA/Type = "OpenGL"
#i:Gr/ViewA/WindowSizeX = 1024
#i:Gr/ViewA/WindowSizeY = 768

Ts/PauseBeforeQuit = "True"

i:Ts/Seed = 2580
i:Ts/NumberOfThreads = -2 # defaults to 1
`

In this case, I'm trying to score the DNA damage of the MyCell/Nucleus, however, when i try this:

s:Sc/DNADamageScorer/Component 				= "MyCell/Nucleus"

I get this error:
DNADamageScorer has unfound component:MyCell/Nucleus

### Version

3.0

### What Operative System (OS) are you running on?

Windows (WSL)

### Relevant log output

```shell
Reading geometry information file...
Reading copy number map file...
Reading chromosome-voxel map file...
Topas is exiting due to a serious error in scoring setup.
DNADamageScorer has unfound component:MyCell/Nucleus
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Hi,

This particular problem occurs because you have no parent defined for `MyCell/Nucleus`. However even if you correctly define the parent as `MyCell` you're still going to get some overlap issues preventing the simulation from running.

I'm not familiar with `TsNucleus`, but unless I am mistaken it needs to be a daughter of the world volume, not of another component within the world. Apologies for not being able to help further.

I will be converting this issue into a discussion.

All the best,
Thongchai
